crates/rpcsrv/src/flight_handler.rs

```diff
@@ -51,7 +53,7 @@ pub struct FlightSessionHandler {
     // We'll want to implement a time based eviction policy, or a max size.
     // We use [`Session`] instead of [`TrackedSession`] because tracked sessions need to be
     // explicitly closed, and we don't have a way to do that yet.
-    sessions: DashMap<String, Arc<Mutex<Session>>>,
+    sessions: DashMap<String, Arc<RwLock<Session>>>,
```

Comment:
```suggestion
    sessions: DashMap<String, Arc<Mutex<Session>>>,
```

I believe pretty strongly in using mutexes over read/write locks when using tokio's implementations as it's assumed you'll be locking over await points. If I had my way, there would be no tokio mutexes in the code base as a believe it points to a suboptimal design, but I also know it's not practical to change it.

Session access is typically serial (in that clients call one method after another and not concurrently) and a normal mutex would have lower overhead in that case. Also tokio mutexes are already slow, and tokio rwlocks are even slower.